### ISSUE TITLE
Would like to see window resizing not confined to lower right corner

### ISSUE THREAD

**Issue by stefanoaz** [XQuartz/XQuartz#169]
I would like to see window behavior in XQuartz conform to both MacOS and most Linux WM's, in terms of more flexible window resizing. I would like to be able to resize a window from all four corners and all four sides. The lower right tab is a relic whose time has past, I hope. Makes it hard to go back and forth between environments when XQuartz's behavior is so different from the standard everywhere else.

**Comment by dpeterc** [comment]
That would be wonderful.
The white resize square also covers some of the application window's contents, and conflicts the little style the X11 applications might have.

**Comment by ianfixes** [comment]
This also ends up being important in cases where initial window sizes are slightly taller than the available screen.  That's likely related to #92

**Comment by jeremyhu** [comment]
Yeah, the best solution would be to change XQuartz into a compositing window manager such that we had native NSWindows hosting the X11 window contents.

**Comment by dillfitz** [comment]
Any progress on this? The current implementation of the window resize slows down my workflow quite a bit since switching from Ubuntu.

**Comment by jeremyhu** [comment]
There's not really anyone working on this any more.  I take a look at it for maybe a days every few years, but that's not enough to make a meaningful dent, especially for an involved project like this specific fix would be.

**Comment by sohale** [comment]
I need this desperately as I cannot use `git-cola` properly. Here is my take:

## Title

Impossible to access window bottom for resizing when X11 window is taller than visible macOS screen due to Dock/Menu Bar occlusion

## Description

On macOS (with XQuartz), some X11 applications (e.g., git-cola) open windows that use the full available screen height. However, if the window height includes the area under the macOS Dock (the horizontal bar at the bottom of the screen), the bottom of the window (including the resize handle) becomes inaccessible. This makes it impossible to resize the window vertically, since the only way to resize is from the bottom-right corner, which is hidden behind the Dock.

Even when moving the window (by dragging title bar or some Option key + drag), it cannot be moved high (up) enough to expose the bottom edge: Because it cannot be moved further up (the macOS menu bar at the top prevents further upward movement, but it could be the screen's top end). As a result of a choice by git-cola to use maximal height -- which is a sensible choice. This creates a situation where the window is "stuck" at a size that cannot be changed by the user. Hence, some content or controls may be permanently inaccessible. In this particular case, I cannot see or select git-diff modified lines shown at the bottom of `git-cola`'s window.


## Why this matters:

Users cannot access or resize X11 windows that are too tall.
Some applications (like `git-cola`) do not provide alternative resize mechanisms or geometry settings.
This is not just an inconvenience: it can make parts of applications (such as buttons or modified file lists) completely unusable. The way I "generalise" this (which is key in any issue), is that if an independently-developed floating UI-element (here, MacOC's Dock) occludes part of the window, that is the resize/move handle, the user is doomed!

## Proposed solution:

Add a "Resize" option to the XQuartz Window menu, similar to how "Move" works, allowing users to enter a resize mode (perhaps with a modifier key or menu item) and resize the window from any edge using the keyboard or mouse, regardless of whether the window edge is visible.
Alternatively, provide a way to force all new X11 windows to fit within the visible macOS screen area, above the Dock and below the menu bar.

This UI reachability paradox creates a geometric "dead zone" —the window can be moved, but never enough to access the resize handle, due to constraints independent of "application"-independent  (here, from macOS "Dock", but as a generalizaion it could be from any floating UI elements that cannot be moved). A keyboard-based or menu-based resize mode would resolve this.

## Steps to reproduce:

Open an X11 app (e.g., git-cola) via XQuartz on macOS.
The app opens a window that is as tall as the screen, with the bottom edge under the Dock.
Try to resize the window vertically—only the bottom-right corner allows resizing, but it is inaccessible.
Try to move the window up—macOS menu bar prevents moving it high enough to expose the bottom.
System info:

macOS version: 15.3.2
XQuartz version: 2.8.5 (xorg-server 21.1.6)
Example app: `git-cola` (but affects others)
Dock position: bottom (default)

**Comment by robohack** [comment]
Yeah, using mixed X11 and Aqua workspaces has always seemed to me like a great way to ask for trouble.

If you run XQuartz in Full Screen mode, possibly on a dedicated desktop workspace as well, and you run a normal X11 window manager (e.g. I use ctwm) within it, then you have full control over all X11 applications and their windows just as if you were running them on a native Xserver.

One more advantage to running in Full Screen mode is that if you have a "retina" display then you can get X11 to take full advantage of the resolution instead of having to be restricted to Aqua's limited scaled display resolution.  For example on an Apple Pro Display XDR you can run XQuartz at the full 6016x3384 resolution while Aqua is limited to 1/2 of that each way, so 4 times the pixels.  Since XQuartz can't make hidden use of the extra resolution for anti-aliasing in the way Aqua can, this is necessary to get better image and font display in X11 applications.  I think I've posted elsewhere in another issue about how I do this.

**Comment by Dave-Allured** [comment]
Above, @sohale said:
> some content or controls may be permanently inaccessible

Not quite.  Here is a simple escape mechanism.  Indeed it took me a while to notice this.

Simply click on the green "maximize" button next to the yellow button in the upper left.  Alternatively, use the Apple top nav bar menu: **Window/Maximize**.

This should re-fit your current X11 window to the full screen, except it is the macOS fake full screen, in which the Apple top nav bar and the X11 frame border are still exposed.

Notice:  The resizing handle in the lower right border is now visible!  Simply grab and resize to whatever you need!  Problem solved!

My experience with this is limited to nedit and X11 terminal windows.  It is possible that some X11 apps don't work like this, but my hunch is that they all do.

All that notwithstanding, I still fully support the original request to provide resizing on all borders.

**Comment by Dave-Allured** [comment]
> Forgive the stupid question, but can't you just move the dock to another edge if this is so critical?

This does not help when I have a tall X11 window on a large monitor, and then take my laptop away so that I have only the smaller laptop screen.  XQuartz "helpfully" remembers the exact size of my original window, so the resize handle is now way off the bottom of the smaller screen.  This has nothing to do with the dock position.  Nothing will bring back the handle -- except for the Maximize trick that I just described.

**Comment by robohack** [comment]
> This does not help when I have a tall X11 window on a large monitor, and then take my laptop away so that I have only the smaller laptop screen. XQuartz "helpfully" remembers the exact size of my original window, so the resize handle is now way off the bottom of the smaller screen. This has nothing to do with the dock position. Nothing will bring back the handle -- except for the Maximize trick that I just described.

There's apparently a package called `autorandr` ([https://github.com/phillipberndt/autorandr](https://github.com/phillipberndt/autorandr) is one implementation) that can help manage changes in display hardware for X11, but I've no idea how well it may or may not work on macos with XQuartz.
